File: .claude/skills/calma/assets/lang/r/gen_fixture.R
# Calma cross-language fixture (R). Emits the shared deterministic return series; prints total_return.
dir.create("runs", showWarnings=FALSE)
r <- sapply(0:251, function(i) 0.0005 + 0.03*sin(i*0.2))
con <- file("runs/returns.csv","w"); writeLines("daily_return", con)
for (x in r) writeLines(format(x, digits=17), con)
close(con)
cat(sprintf("total_return=%.17g\n", prod(1+r)-1))

File: .claude/skills/calma/assets/lang/rust/gen_fixture.rs
// Calma cross-language fixture (Rust).
use std::fs; use std::io::Write;
fn main(){
  fs::create_dir_all("runs").unwrap();
  let mut f = fs::File::create("runs/returns.csv").unwrap();
  writeln!(f, "daily_return").unwrap();
  let mut acc = 1.0f64;
  for i in 0..252 { let x = 0.0005 + 0.03*((i as f64)*0.2).sin(); writeln!(f, "{}", x).unwrap(); acc *= 1.0+x; }
  println!("total_return={}", acc-1.0);
}
